### PR TITLE
Add support for passing custom headers in Spring Cloud Config

### LINKED ISSUE #12099: Add support for HTTP headers in Spring Cloud Config Client extension
**Description**
The feature request is to add a new configuration property to support generic HTTP headers that would be used on the HTTP requests from the Spring Cloud config client to the Spring config server. This could be used to pass JWT credentials in an Authorization header, for example. 

This is currently part of Spring's own config client implementation, accessible through the following property: 
```spring.cloud.config.headers```

**Implementation ideas**
In Quarkus, the property name could be:
```quarkus.spring-cloud-config.headers```

taking a Map<String, String> type, as in the Spring implementation.

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #12099

The first commit fixes bootstrap config in order to enable `Map` support for that Config.

### CODE REVIEW

**Comment by geoand** [comment]
I would like a review from @dmlloyd on the first commit and @gastaldi or @gsmet for the second one

File: extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/SpringCloudConfigClientConfig.java [hunk 1]
```diff
@@ -96,7 +97,13 @@
...
      */
     @ConfigItem(defaultValue = "false")
-    boolean trustCerts;
+    public boolean trustCerts;
+
+    /**
+     * Custom headers to pass the Spring Cloud Config Server when performing the HTTP request
+     */
+    @ConfigItem
+    public Map<String, String> headers;
```
Comment (gastaldi) [inline extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/SpringCloudConfigClientConfig.java:106]: The HTTP spec [RFC 2616][1] allows multiple values for a single HTTP header: 

> Multiple message-header fields with the same field-name MAY be present in a message if and only if the entire field-value for that header field is defined as a comma-separated list [i.e., #(values)]. It MUST be possible to combine the multiple header fields into one "field-name: field-value" pair, without changing the semantics of the message, by appending each subsequent field-value to the first, each separated by a comma. The order in which header fields with the same field-name are received is therefore significant to the interpretation of the combined field value, and thus a proxy MUST NOT change the order of these field values when a message is forwarded.

What this means is that you can send multiple values in a single header by concatenating them with commas. Do you want to support that here or that wouldn't matter?

[1]: http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2

Comment (geoand) [reply extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/SpringCloudConfigClientConfig.java:106]: Yeah I know, but I really don't think it's necessary. Best keep the common case easy to use.

**Review by gastaldi — APPROVED** [review]
LGTM. Looks like the headers are just being used to send to Spring Cloud so it doesn't matter if they are multiple or not

File: core/deployment/src/main/java/io/quarkus/deployment/configuration/RunTimeConfigurationGenerator.java [hunk 2]
```diff
@@ -1630,4 +1640,16 @@ private static boolean isFieldEligibleForDirectAccess(ClassDefinition.ClassMembe
                 && Modifier.isPublic(classMember.getEnclosingDefinition().getConfigurationClass().getModifiers())
                 && Modifier.isPublic(classMember.getField().getType().getModifiers());
     }
+
+    private enum Type {
+        BuildTime("si"),
+        Bootstrap("bs"),
+        Runtime("rt");
```
Comment (gastaldi) [inline core/deployment/src/main/java/io/quarkus/deployment/configuration/RunTimeConfigurationGenerator.java:1647]: Uppercase them?

Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/configuration/RunTimeConfigurationGenerator.java:1647]: Yeah I guess most of our enums are upper case, so I'll fix it

Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/configuration/RunTimeConfigurationGenerator.java]: Fixed

**Comment by geoand** [comment]
@dmlloyd mind taking a quick look at the first commit please?

Thanks

**Comment by dmlloyd** [comment]
First commit seems OK to me.

**Pull request merged**